src/MainAppRouter.js

```diff
@@ -53,11 +54,8 @@ export default function MainAppRouter({ api, setUser, hasExtension }) {
 
     if (redirectLink !== null) {
       window.location.href = redirectLink;
-    } else if (
-      window.location.href.indexOf("create_account") > -1 &&
-      !hasExtension && isSupportedBrowser()
-    ) {
-      history.push("/install_extension");
+    } else if (window.location.href.indexOf("create_account") > -1) {
+      history.push("/select_interests");
```

Comment:
also, based on our call, ideally create_account page should send the user to the `/select_interests` rather than doing this once off special routing here. if you can handle this and the `PrivateRoute` observation of Tiago below, I think this is ready to be merged!

Reply:
I agree. It's better to send the user directly from the registration page to the interest selection page. However, I have not implemented this yet. To proceed, I need to understand better the `handleSuccessfulSignIn` function defined in the `MainAppRouter` and its dependencies. <br>
**Would you mind if we had a discussion about it?** 
<br>
So far, I have observed that `handleSuccessfulSignIn`: 
- takes **userInfo** as a parameter
- it's shared between the `SignIn` and `CreateAccount` pages. 
- In both cases, it is sent as a **prop** called `signInAndRedirect`, which takes two arguments when called: userInfo, history

To my understanding, `handleSuccessfulSignIn` is responsible for:
- setting a session
- getting user preferences from the API
- saving user info to cookies
- setting the user
- redirecting using `history`

Reply:
**An update:**

I just discussed this with @tfnribeiro, and I have a better understanding of what `handleSuccessfulSignIn` does and an Idea of how to proceed. <br>
**I will be working on the following:**
* Updating the props' name of the `SignIn` and `CreateAccount` components so that it’s clear that they point to the `handleSuccessfulSignIn` function
* Making the routing process separated from this function
* Creating a reusable function that handles redirecting, e.g. `redirect(link)`
* Handling redirecting from the `create_account` page component, not from the `MainAppRouter` component

Reply:
@igawaclawska please do ping me when this is done so I can merge it in the master then!

Reply:
Hi @mircealungu , all the things mentioned above are implemented now. 
<br>
**Additionally:** 

- Basic **redirect** functionality has been extracted and placed in the `utils/routing` directory as a reusable, standalone function:
```
export default function redirect(linkToRedirect) {
  window.location.href = linkToRedirect;
}
```
- A redirect functionality that supports existing redirect links has been extracted into a reusable custom hook, called `useRedirectLink`:

```
/* This custom hook is meant to be used for redirecting
on pages that should support redirect links */
export default function useRedirectLink() {
  const [existingRedirectLink, setExistingRedirectLink] = useState(null);

  //check for existing redirect links
  useEffect(() => {
    const queryParameters = new URLSearchParams(window.location.search);
    setExistingRedirectLink(queryParameters.get("redirectLink"));
  }, []);

  function handleRedirect(linkToRedirect) {
    existingRedirectLink
      ? redirect(existingRedirectLink)
      : redirect(linkToRedirect);
  }
  return { handleRedirect };
```

- The `CreateAccount` page supports **basic** redirect functionality because it should always send the user to the **onboarding** pages - `\select_interests` in this case.

- The `SignIn` page **supports redirect links**. If a redirect link exists, the user is sent there, otherwise they are sent to the `\articles`.